astropy/cosmology/flrw/base.py

```diff
@@ -194,14 +194,48 @@ def critical_density(self, z: Quantity | ArrayLike) -> Quantity:
         return self.critical_density0 * self.efunc(z) ** 2
 
 
+class _BaryonComponent:
+    """The cosmology has attributes and methods for the baryon density."""
+
+    Ob0: float
```

Comment:
Why is `Ob0` not moved to the new mixin?

Reply:
This will allow, e.g. models without `Ob0` as a cosmological parameter.

Reply:
My thinking is that if `Ob(z)` is well defined then `Ob0=Ob(0)` must also be well defined, so why can't this mixin handle both?

Reply:
`Ob0=Ob(0)` is well defined as `Ob0`. In (nearly?) all model formulations `Ob0` is either
1. Not a part of the cosmology, e.g. a Lambda-only universe or a radiation-only universe. 
2. A settable parameter of the cosmology
3. A derived parameter of the cosmology.

So I don't want to make `Ob0` "defined" in the sense of a "Parameter" as it is on the FLRW objects.

Any class that subclasses `_BaryonComponent` is in category 2 or 3, but if it's a derived parameter then it is vastly preferable to make `Ob0` a `~property`, not a "astropy.cosmology.Parameter"